packages/features/bookings/Booker/utils/query-param.ts

```diff
@@ -2,13 +2,19 @@ export const updateQueryParam = (param: string, value: string | number) => {
   if (typeof window === "undefined") return;
 
   const url = new URL(window.location.href);
+  if (url.searchParams.get(param) === value) return;
+
   if (value === "" || value === "null") {
-    url.searchParams.delete(param);
+    removeQueryParam(param);
+    return;
   } else {
     url.searchParams.set(param, `${value}`);
   }
-
-  window.history.pushState({}, "", url.href);
+  if (param == "slot") {
```

Comment:
For the desired behavior, we  need to smartly choose between `pushState` and `replaceState` so that the browser history is created the way we want.  This is  in-line with the change that @Udit-takkar requested earlier.